### DISCUSSION TITLE
Pocketbases Authentication And Database

### DISCUSSION THREAD

**Discussion by AxewBoTX** [pocketbase/pocketbase#3200] [Q&A]
Heyy,  I am a beginner to this database and authentication subject and pocketbase seemed the best option to me as it is fully self hostable, but there are a few problems that I am running into and a few things I really need help with.<br>
1. Is there any full step by step guide for authentication for absolute beginners? 
2. I am using the basic collection of users that pocketbase already creates for you, I know how to authenticate with current existing users in that collection but I dont know how to add users like signup them to that collection. Can you please provide help with that?
3. How can I keep the currently logged in user logged in even if he closes the website?

Please it would be great help if you can answer all these questions separately.
Thanks for your effors!

**Comment by ganigeorgiev** [comment]
1. The question is too vague and I'm not sure that I understand what is not clear and what steps are expected. Currently we have only https://pocketbase.io/docs/authentication/.

2. Registering a new user is no much different than creating a new record in the users collection. You can find a copy-paste example in the "API Preview" section for your "users" collection from the Admin UI.

3.  A request is considered authenticated as long as it has `Authorization: TOKEN` with it. If you are using the [JS SDK](https://github.com/pocketbase/js-sdk) or [Dart SDK](https://github.com/pocketbase/dart-sdk), once you authenticate this header will be automatically added with any following requests. The token is kept in `pb.authStore`. The JS SDK by default uses `localStorage` meaning that the auth data will be automatically loaded the next time the user open your website (_as long as the client is initiated client-side, in the browser_). For more details about the JS AuthStore you can check https://github.com/pocketbase/js-sdk#auth-store.
